pkg/ebpf/tracee.go

```diff
@@ -783,6 +784,29 @@ func (t *Tracee) populateBPFMaps() error {
 				return err
 			}
 		}
+		if e == PrintSyscallTableEventID {
+			hookedSyscallsMap, err := t.bpfModule.GetMap("syscalls_to_check_map")
+			if err != nil {
+				return err
+			}
+			/* the hookedSyscallsMap store first the syscall table address then the syscall numbers, like that:
+			 * [sys_call_table symbol address][syscall num #1][syscall num #2][syscall num #3]...
+			 * with that, we can fetch the syscall address by accessing the syscall table in the syscall number index
+			 */
+			key := int(0)
+			arch := t.config.OsConfig.GetOSReleaseFieldValue(helpers.OS_ARCH)
+			syscallList := getSyscallList(arch)
+			// the map should look like [syscall number 1][syscall number 2][syscall number 3]...
+			for idx, val := range syscallList {
+				key = idx
+				errs = append(errs, hookedSyscallsMap.Update(unsafe.Pointer(&(key)), unsafe.Pointer(&val)))
+			}
+			for _, err := range errs {
+				if err != nil {
+					return err
+				}
+			}
```

Comment:
I don't think you need this errs map as you're in a loop, and can break anytime in case of an error.
Just do:
```
err = hookedSyscallsMap.Update(unsafe.Pointer(&(key)), unsafe.Pointer(&val))
if err != nil {
	return err
}
```